dask/dataframe/core.py

```diff
@@ -4289,6 +4289,11 @@ def is_monotonic_increasing(self):
     def is_monotonic_decreasing(self):
         return super().is_monotonic_decreasing
 
+    @derived_from(pd.Index)
+    def memory_usage(self, deep=False):
+        result = self.map_partitions(M.memory_usage, deep=deep, enforce_metadata=False)
+        return delayed(sum)(result.to_delayed())
```

Comment:
This is mostly just copying the existing `Series.memory_usage` implementation. This works, but it'd be nice to return a `dd.Scalar` instead of a `Delayed` object so we pick up all the default `DataFrame` graph optimizations. Will try to push up a commit to handle that